build.sh

```diff
@@ -0,0 +1,548 @@
+VERBOSE=0        # Verbose output flag
+QUICK=0          # Quick test mode (subset of tests)
+
+# Test configuration (0=disabled, 1=enabled)
+BUILD_TESTS=0    # Build test binaries
+RUN_UNIT_TESTS=0 # Run C/C++ unit tests
+RUN_RUST_TESTS=0 # Run Rust tests
+RUN_PYTEST=0     # Run Python tests
+RUN_ALL_TESTS=0  # Run all test types
+
+#-----------------------------------------------------------------------------
+# Function: parse_arguments
+# Parse command-line arguments and set configuration variables
+#-----------------------------------------------------------------------------
+parse_arguments() {
+  for arg in "$@"; do
+    case $arg in
+      COORD=*)
+        COORD="${arg#*=}"
+        ;;
+      DEBUG=*)
+        DEBUG="${arg#*=}"
+        ;;
+      STATIC=*)
+        STATIC="${arg#*=}"
+        ;;
+      LITE=*)
+        LITE="${arg#*=}"
+        ;;
+      TESTS=*)
+        BUILD_TESTS="${arg#*=}"
+        ;;
+      RUN_TESTS=*)
+        RUN_ALL_TESTS="${arg#*=}"
+        ;;
+      RUN_UNIT_TESTS=*)
+        RUN_UNIT_TESTS="${arg#*=}"
+        ;;
+      RUN_RUST_TESTS=*)
+        RUN_RUST_TESTS="${arg#*=}"
+        ;;
+      RUN_PYTEST=*)
+        RUN_PYTEST="${arg#*=}"
+        ;;
+      TEST=*)
+        TEST_FILTER="${arg#*=}"
+        ;;
+      SAN=*)
+        SAN="${arg#*=}"
+        ;;
+      FORCE=*)
+        FORCE="${arg#*=}"
+        ;;
+      VERBOSE=*)
+        VERBOSE="${arg#*=}"
+        ;;
+      QUICK=*)
+        QUICK="${arg#*=}"
+        ;;
+      SA=*)
+        SA="${arg#*=}"
+        ;;
+      REDIS_STANDALONE=*)
+        REDIS_STANDALONE="${arg#*=}"
+        ;;
+      *)
+        # Pass all other arguments directly to CMake
+        CMAKE_ARGS="$CMAKE_ARGS -D${arg}"
+        ;;
+    esac
+  done
+}
+
+#-----------------------------------------------------------------------------
+# Function: setup_test_configuration
+# Configure test settings based on input arguments
+#-----------------------------------------------------------------------------
+setup_test_configuration() {
+  # If any tests will be run, ensure BUILD_TESTS is enabled
+  if [[ "$RUN_ALL_TESTS" == "1" || "$RUN_UNIT_TESTS" == "1" || "$RUN_RUST_TESTS" == "1" || "$RUN_PYTEST" == "1" ]]; then
+    if [[ "$BUILD_TESTS" != "1" ]]; then
+      echo "Test execution requested, enabling test build automatically"
+      BUILD_TESTS="1"
+    fi
+  fi
+
+  # If RUN_ALL_TESTS is enabled, enable all test types
+  if [[ "$RUN_ALL_TESTS" == "1" ]]; then
+    RUN_UNIT_TESTS=1
+    RUN_RUST_TESTS=1
+    RUN_PYTEST=1
+  fi
+}
+
+#-----------------------------------------------------------------------------
+# Function: setup_build_environment
+# Configure the build environment variables
+#-----------------------------------------------------------------------------
+setup_build_environment() {
+  # Determine build flavor
+  if [ "$SAN" == "address" ]; then
+    FLAVOR="debug-asan"
+  elif [[ "$DEBUG" == "1" ]]; then
+    FLAVOR="debug"
+  else
+    FLAVOR="release"
+  fi
```

Comment:
don't we need coverage or is it redundant/different PR?

Reply:
different PR